### ISSUE TITLE
My Account page form correction

### ISSUE THREAD

**Issue by support27** [woocommerce/woocommerce#22987]
**Describe the bug**
on the my account page for a non logged in user, if "generate a username from the customers email address" is selected, it correctly only shows email address and password to register. However, on the login side, it shows "username or email address". it should only show "Email address". 

If "generate a username from the customers email address" was NOT enabled, then it should show "username or email address".

**Screenshots**

**Expected behavior**
IF "generate a username from the customers email address" SELECTED show "Email address" for login.
IF "generate a username from the customers email address" NOT SELECTED show "Username or email address" for login.

**Isolating the problem (mark completed items with an [x]):**
- [ x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
5.1
![WC_login](https://user-images.githubusercontent.com/48425726/54099480-c877b100-4376-11e9-81e5-940ba224ba75.png)

```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
`
### WordPress Environment ###

Home URL: https://support27.com
Site URL: https://support27.com
WC Version: 3.5.6
Log Directory Writable: ✔
WP Version: 5.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.2.15
PHP Post Max Size: 33 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.45.0
OpenSSL/1.0.1e

SUHOSIN Installed: –
MySQL Version: 5.6.32-78.0-log
Max Upload Size: 33 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.5.6
WC Database Prefix: wp_akqzmhydy9_
MaxMind GeoIP Database: ✔
Total Database Size: 7.74MB
Database Data Size: 6.99MB
Database Index Size: 0.75MB
wp_akqzmhydy9_woocommerce_sessions: Data: 0.01MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_order_items: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_order_itemmeta: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_tax_rates: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_woocommerce_log: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_commentmeta: Data: 0.00MB + Index: 0.01MB
wp_akqzmhydy9_comments: Data: 0.01MB + Index: 0.01MB
wp_akqzmhydy9_et_divi_ab_testing_stats: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_failed_jobs: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_gf_draft_submissions: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_gf_entry: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_gf_entry_meta: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_gf_entry_notes: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_gf_form: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_gf_form_meta: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_gf_form_revisions: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_gf_form_view: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_links: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_mailchimp_carts: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_options: Data: 2.47MB + Index: 0.05MB
wp_akqzmhydy9_postmeta: Data: 0.26MB + Index: 0.12MB
wp_akqzmhydy9_posts: Data: 0.90MB + Index: 0.05MB
wp_akqzmhydy9_queue: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_sm_sessions: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_ta_link_clicks: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_ta_link_clicks_meta: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_termmeta: Data: 0.00MB + Index: 0.01MB
wp_akqzmhydy9_terms: Data: 0.00MB + Index: 0.01MB
wp_akqzmhydy9_term_relationships: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_term_taxonomy: Data: 0.00MB + Index: 0.01MB
wp_akqzmhydy9_usermeta: Data: 0.02MB + Index: 0.02MB
wp_akqzmhydy9_users: Data: 0.00MB + Index: 0.01MB
wp_akqzmhydy9_wc_download_log: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wc_webhooks: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wfblockediplog: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wfblocks7: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wfconfig: Data: 1.18MB + Index: 0.01MB
wp_akqzmhydy9_wfcrawlers: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wffilechanges: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wffilemods: Data: 1.31MB + Index: 0.28MB
wp_akqzmhydy9_wfhits: Data: 0.03MB + Index: 0.01MB
wp_akqzmhydy9_wfhoover: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wfissues: Data: 0.01MB + Index: 0.01MB
wp_akqzmhydy9_wfknownfilelist: Data: 0.66MB + Index: 0.09MB
wp_akqzmhydy9_wflivetraffichuman: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wflocs: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wflogins: Data: 0.01MB + Index: 0.00MB
wp_akqzmhydy9_wfnotifications: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wfpendingissues: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wfreversecache: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wfsnipcache: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_wfstatus: Data: 0.12MB + Index: 0.05MB
wp_akqzmhydy9_wftrafficrates: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_yoast_seo_links: Data: 0.00MB + Index: 0.00MB
wp_akqzmhydy9_yoast_seo_meta: Data: 0.00MB + Index: 0.00MB

### Post Type Counts ###

attachment: 62
custom_css: 1
custom-css-js: 1
customize_changeset: 30
et_pb_layout_stashed: 8
jetpack_migration: 2
jp_img_sitemap: 1
jp_sitemap: 1
jp_sitemap_master: 1
layout: 2
nav_menu_item: 12
page: 20
post: 3
product: 10
product_variation: 32
revision: 78
scheduled-action: 12
shop_coupon: 1
shop_order: 5
shop_subscription: 1
thirstylink: 1
ticket: 1
ticket_history: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (19) ###

ManageWP - Worker: by GoDaddy – 4.7.5
Akismet Anti-Spam: by Automattic – 4.1.1
Awesome Support: by Awesome Support Team – 5.8.0
WPBakery Page Builder: by Michael M - WPBakery.com – 5.7
Mailchimp for WooCommerce: by Mailchimp – 2.1.14 – Not tested with the active version of WooCommerce
My Custom Functions: by Space X-Chimp – 4.35
myStickymenu: by m.r.d.a – 2.0.6
Regenerate Thumbnails: by Alex Mills (Viper007Bond) – 3.1.0
Search Engine Visibility: by GoDaddy – 0.5
Simple Custom CSS: by John Regan
Danny Van Kooten – 4.0.1

Tawk.to Live Chat: by Tawkto – 0.3.6
Under Construction: by WebFactory Ltd – 3.31
WC Password Strength Settings: by Daniel Santoro – 2.0.2 – Not tested with the active version of WooCommerce
WooCommerce Stripe Gateway: by WooCommerce – 4.1.14
WooCommerce Subscriptions: by Prospress Inc. – 2.5.2
WooCommerce: by Automattic – 3.5.6
Wordfence Security: by Wordfence – 7.2.3
Yoast SEO: by Team Yoast – 9.7
WP File Manager: by mndpsingh287 – 4.4

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Shop base: #166 - /shop
Cart: #167 - /cart
Checkout: #168 - /checkout
My account: #169 - /my-account
Terms and conditions: #297 - /terms

### Theme ###

Name: Divi
Version: 3.19.18
Author URL: http://www.elegantthemes.com
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
Subscriptions Live URL: https://support27.com
Subscription Statuses: wc-active: 1
WooCommerce Account Connected: ✔ Yes
Active Product Key: ✔ Yes
Report Cache Enabled: ✔ Yes
Cache Update Failures: ✔ 0 failure

### Store Setup ###

Country / State: United States (US) — Nevada

### Subscriptions by Payment Gateway ###

Stripe: wc-active: 1

### Payment Gateway Support ###

Stripe: products
refunds
tokenization
add_payment_method
subscriptions
subscription_cancellation
subscription_suspension
subscription_reactivation
subscription_amount_changes
subscription_date_changes
subscription_payment_method_change
subscription_payment_method_change_customer
subscription_payment_method_change_admin
multiple_subscriptions
pre-orders

`

**Comment by kloon** [comment]
This is not a bug, WordPress allows you to log in with either your registered email address or username. The option to generate a username from email is simply functionality we added to speed up the registration process so the customer does not have to choose a username. Their email address is still tied to their account and as such, they can log in with that as well as the generated username.

WordPress requires that when you register a user account you specify an email as well as a username.

**Comment by support27** [comment]
It's not necessarily about WordPress. For the woocommerce my account page, it'd be nice if the login side, when a separate username is not created it specifies just email rater than username OR email since in that case they are identical.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by kloon** [comment]
You are welcome to override the template and change the wording yourself, but as I stated it mentions both username and email as those are the two required fields when you register a new WordPress account and since WooCommerce customers uses WordPress accounts that is why we state both as even with the option specifying you to auto create a username you can still log in with either the auto created username as well as the email address.

For details on overriding template please see https://docs.woocommerce.com/document/template-structure/

**Comment by support27** [comment]
Awesome! Thanks

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by support27** [comment]
Hello,

 

Ran into the issue where I made the change but of course with the plugin update the changes were overridden. I am now trying to accomplish this using a hook but not having much success. 

 

add_filter( 'woocommerce_login_form_start', 'remove_username_or' );

 

function remove_username_or( $fields ) {

// Your code

return $fields['username'] = 'Email address';

}

 

Any ideas where I am going wrong? Help is much appreciated.

 

Best Regards,

 

support27

2691 E Hacienda Ave

Las Vegas, NV 89120

 

www.support27.com

info@support27.com

+1 (559) 362-0194
